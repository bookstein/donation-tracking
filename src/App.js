import React, { Component } from 'react'
import './App.css'
import ProfileContainer from './containers/profileContainer'

class App extends Component {
  render() {
    return <ProfileContainer />
  }
}

export default App
