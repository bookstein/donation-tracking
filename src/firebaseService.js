import fb from './firebase'

export const listenForUpdates = (refName, cb) => {
  let myRef = fb
    .database()
    .ref(refName)
    .orderByKey()

  myRef.on('child_added', cb)
  myRef.on('child_removed', cb)
}

export const pushToDatabase = (refName, data) => {
  const myRef = fb
    .database()
    .ref(refName)
    .push()
  myRef.set(data)
  // for some reason I can only use `set`
  // `push` seems to append to the last item in the collection
  console.log('view your new data at ', myRef.toString())
}

export const removeFromDatabase = (refName, key) => {
  const myRef = fb.database().ref(refName)
  myRef
    .child(key)
    .remove()
    .then(function() {
      console.log('Remove succeeded.')
    })
    .catch(function(error) {
      console.log('Remove failed: ' + error.message)
    })
}
