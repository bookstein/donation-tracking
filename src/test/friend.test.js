import React from 'react'
import ReactDOM from 'react-dom'
import App from '../App'
import Friend from '../components/Friend'

// get testing tools
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import { removeFromDatabase } from '../firebaseService'

const removeFriend = jest.fn()
const PROPS = {
  removalKey: 'testKey',
  friendTag: 'testTag',
  friendName: 'testName',
  removeFriend,
}

// happy path: component renders (jest snapshot)
it('renders without crashing', () => {
  const component = renderer.create(<Friend {...PROPS} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// simulate user behavior
// click X
it('calls the `remove` callback on click', () => {
  const component = shallow(<Friend {...PROPS} />)
  component.find('.Friend__X').simulate('click')
  expect(removeFriend).toHaveBeenCalled()
})
