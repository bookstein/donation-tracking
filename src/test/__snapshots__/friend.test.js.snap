// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="Friend"
>
  <div>
    <h1>
      testName
    </h1>
    <p
      className="Friend__Tags"
    >
      Tags:
      <span
        className="Friend__Tag"
      >
        testTag
      </span>
    </p>
    <div
      className="Friend__Info"
    >
      <h4>
        Donations:
      </h4>
      <p>
        <label>
          Pledged to donate?
        </label>
        <input
          className="Friend__formField"
          type="checkbox"
        />
      </p>
      <p>
        <label>
          Amount pledged:
        </label>
        <span>
          $
        </span>
        <input
          className="Friend__formField"
          placeholder=""
          type="text"
        />
      </p>
    </div>
  </div>
  <div
    className="Friend__X"
    onClick={[Function]}
  >
    X
  </div>
</div>
`;
